modules/vault/src/main/java/org/apache/ignite/internal/vault/common/Watch.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.vault.common;
+
+import java.util.Comparator;
+import org.jetbrains.annotations.Nullable;
+
+public class Watch {
```

Comment:
Why not to reuse watch abstraction from DMS?

Reply:
`WatchListener` depends on `WatchEvent`, which depends on `Entry`, `Entry` has `revision` and will have `updateCounter` in the future. Seems that it is not suitable for `Vault` as far as `Vault` don't need to have such fields.